internal/localdebug/local_ide_debug.go

```diff
@@ -0,0 +1,263 @@
+    <RunnerSettings RunnerId="Debug">
+      <option name="DEBUG_PORT" value="%d" />
+      <option name="LOCAL" value="false" />
+    </RunnerSettings>
+    <method v="2" />
+  </configuration>
+</component>`
+
+const golangDebugConfig = `<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="%s" type="GoRemoteDebugConfigurationType" factoryName="Go Remote" port="%d">
+    <option name="disconnectOption" value="LEAVE" />
+    <disconnect value="LEAVE" />
+    <method v="2" />
+  </configuration>
+</component>`
+
+func init() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT)
+	signal.Notify(c, syscall.SIGTERM)
+	go func() {
+		<-c
+		lock.Lock()
+		defer lock.Unlock()
+		for p := range paths {
+			println(p)
+			_ = os.Remove(p)
+		}
+
+	}()
+}
+
+var lock = sync.Mutex{}
+var paths = map[string]bool{}
+
+// SyncIDEDebugIntegrations will sync the local IDE debug configurations for the given project path.
+// This is a bit of a hack to prove out the concept of local debugging.
+func SyncIDEDebugIntegrations(cxt context.Context, projectPath string, ports map[string]*DebugInfo) {
+	lock.Lock()
+	defer lock.Unlock()
+	if projectPath == "" {
+		return
+	}
+	handleIntellij(cxt, projectPath, ports)
+	handleVSCode(cxt, projectPath, ports)
+}
+
+func handleIntellij(cxt context.Context, path string, ports map[string]*DebugInfo) {
+	logger := log.FromContext(cxt)
+	ideaPath := findIdeaFolder(path)
+	if ideaPath == "" {
+		return
+	}
+	runConfig := filepath.Join(ideaPath, "runConfigurations")
+	err := os.MkdirAll(runConfig, 0600)
+	if err != nil {
+		logger.Errorf(err, "could not create runConfigurations directory")
+		return
+	}
+	entries, err := os.ReadDir(runConfig)
+	if err != nil {
+		logger.Errorf(err, "could not create runConfigurations directory")
+		return
+	}
+	for _, entry := range entries {
+		if strings.HasPrefix(entry.Name(), "FTL.") && strings.HasSuffix(entry.Name(), ".xml") {
+			debugInfo := ports[entry.Name()[4:(len(entry.Name())-4)]]
+			if debugInfo == nil {
+				// old FTL entry, remove it
+				path := filepath.Join(runConfig, entry.Name())
+				_ = os.Remove(path)
+			}
+		}
+	}
+	for k, v := range ports {
+		if v.Language == "java" || v.Language == "kotlin" {
+			name := filepath.Join(runConfig, "FTL."+k+".xml")
+			err := os.WriteFile(name, []byte(fmt.Sprintf(jvmDebugConfig, "FT𝝺 JVM - "+k, v.Port, v.Port)), 0600)
+			paths[name] = true
+			if err != nil {
+				logger.Errorf(err, "could not create FTL Java Config")
+				return
+			}
+		} else if v.Language == "go" {
+			name := filepath.Join(runConfig, "FTL."+k+".xml")
+			err := os.WriteFile(name, []byte(fmt.Sprintf(golangDebugConfig, "FT𝝺 GO - "+k, v.Port)), 0600)
+			paths[name] = true
+			if err != nil {
+				logger.Errorf(err, "could not create FTL Go Config")
+				return
+			}
+		}
+	}
+}
+
+// findIdeaFolder recurses up the directory tree to find a .idea folder.
+func findIdeaFolder(startPath string) string {
```

Comment:
Duplicate code to that in the VSCode one, parameterise on the relative path you're looking for.